dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotRestHighLevelClientProvider.java

```diff
@@ -0,0 +1,354 @@
+                        "([a-z0-9+/=\\r\\n]+)" +                       // Base64 text
+                        "-+END\\s+.*PRIVATE\\s+KEY[^-]*-+",            // Footer
+                CASE_INSENSITIVE);
+
+        private PemReader() {
+
+        }
+
+        private static KeyStore loadTrustStore(File certificateChainFile)
+                throws IOException, GeneralSecurityException {
+            KeyStore keyStore = KeyStore.getInstance("JKS");
+            keyStore.load(null, null);
+
+            List<X509Certificate> certificateChain = readCertificateChain(certificateChainFile);
+            for (X509Certificate certificate : certificateChain) {
+                X500Principal principal = certificate.getSubjectX500Principal();
+                keyStore.setCertificateEntry(principal.getName("RFC2253"), certificate);
+            }
+            return keyStore;
+        }
+
+        private static KeyStore loadKeyStore(File certificateChainFile, File privateKeyFile,
+                Optional<String> keyPassword)
+                throws IOException, GeneralSecurityException {
+            PKCS8EncodedKeySpec encodedKeySpec = readPrivateKey(privateKeyFile, keyPassword);
+            PrivateKey key;
+            try {
+                KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+                key = keyFactory.generatePrivate(encodedKeySpec);
+            } catch (InvalidKeySpecException ignore) {
+                KeyFactory keyFactory = KeyFactory.getInstance("DSA");
+                key = keyFactory.generatePrivate(encodedKeySpec);
+            }
+
+            List<X509Certificate> certificateChain = readCertificateChain(certificateChainFile);
+            if (certificateChain.isEmpty()) {
+                throw new CertificateException(
+                        "Certificate file does not contain any certificates: "
+                                + certificateChainFile);
+            }
+
+            KeyStore keyStore = KeyStore.getInstance("JKS");
+            keyStore.load(null, null);
+            keyStore.setKeyEntry("key", key, keyPassword.orElse("").toCharArray(),
+                    certificateChain.stream().toArray(
+                            Certificate[]::new));
+            return keyStore;
+        }
+
+        private static PKCS8EncodedKeySpec readPrivateKey(File keyFile,
+                Optional<String> keyPassword)
+                throws IOException, GeneralSecurityException {
+            String content = readFile(keyFile);
+
+            Matcher matcher = KEY_PATTERN.matcher(content);
+            if (!matcher.find()) {
+                throw new KeyStoreException("found no private key: " + keyFile);
+            }
+            byte[] encodedKey = base64Decode(matcher.group(1));
+
+            if (!keyPassword.isPresent()) {
+                return new PKCS8EncodedKeySpec(encodedKey);
+            }
+
+            EncryptedPrivateKeyInfo encryptedPrivateKeyInfo = new EncryptedPrivateKeyInfo(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'encryptedPrivateKeyInfo' could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=3712629)